elastic/kibana
# Removed scripted sort from Task Manager's update by query

In this PR, I'm removing the script that is required for updateByQuery to work and replacing it with a denormalized field. Each create or update to a task document, task manager will compute `claimAt` to match what the script would have done (`retryAt || runAt`). At the beginning, task documents won't have this field but after their next update, they will. I have added special handling for this in the sort by making documents without `claimAt` set, show up first in the result set. This will only happen one time per document because they will have the field set on the next update.

## To verify

Scenario 1: Create alerting rules on a fresh install, ensure they are running as expected.

Scenario 2: Create alerting rules on main (or a version < 8.10), change to this branch and ensure rules are running as expected.

Scenario 3: Create a rule that times out and ensure the retry logic cancels the task and re-runs it at the next interval. You can shut down Kibana to cause a time out and add delays within any rule type execution code to facilitate the timing of shutting down Kibana. After it's shut down, you can restart and see if the rule starts again after 5 to 10 minutes (whatever value is within `retryAt`).
